data/transactions/logic/assembler.go

```diff
@@ -1275,6 +1462,21 @@ func (ops *OpStream) assemble(fin io.Reader) error {
 
 	// TODO: warn if expected resulting stack is not len==1 ?
 	ops.resolveLabels()
+	reverseLabels := make(map[int]string)
+	for key, val := range ops.labels {
+		reverseLabels[val] = key
+	}
```

Comment:
Perhaps `reverseLabels` should be in the ops struct, and setup inside of `resolveLabels`.